sklearn/calibration.py

```diff
@@ -242,6 +245,15 @@ class CalibratedClassifierCV(ClassifierMixin, MetaEstimatorMixin, BaseEstimator)
     array([[0.936..., 0.063...]])
     """
 
+    _parameter_constraints = {
+        "estimator": [BaseEstimator, None],
```

Comment:
```suggestion
        "estimator": [HasMethods(["fit", "predict_proba", "decision_function"]), None],
```

Reply:
In addition, `_get_prediction_method`, we should not have to raise a `RuntimeError` now.
It should be handled with the validation.

Reply:
HasMethods requires all methods to be implemented but here only one of predict_proba or decision_function needs to be implemented. We can do that by having 2 constraints. I directly made the change